src/xrpld/app/tx/detail/Transactor.cpp

```diff
@@ -486,35 +495,100 @@ Transactor::apply()
 NotTEC
 Transactor::checkSign(PreclaimContext const& ctx)
 {
+    // Ignore signature check on batch inner transactions
+    if (ctx.tx.isFlag(tfInnerBatchTxn))
+    {
+        // Defensive Check: These values are also checked in Batch::preflight
+        if (ctx.tx.isFieldPresent(sfTxnSignature) ||
+            !ctx.tx.getSigningPubKey().empty() ||
+            ctx.tx.isFieldPresent(sfSigners))
+        {
+            return temINVALID_FLAG;  // LCOV_EXCL_LINE
+        }
+        return tesSUCCESS;
+    }
+
+    auto const idAccount = ctx.tx.getAccountID(sfAccount);
+
     // If the pk is empty, then we must be multi-signing.
     if (ctx.tx.getSigningPubKey().empty())
-        return checkMultiSign(ctx);
-
-    return checkSingleSign(ctx);
-}
+    {
+        STArray const& txSigners(ctx.tx.getFieldArray(sfSigners));
+        return checkMultiSign(ctx.view, idAccount, txSigners, ctx.j);
+    }
 
-NotTEC
-Transactor::checkSingleSign(PreclaimContext const& ctx)
-{
-    // Check that the value in the signing key slot is a public key.
+    // Check Single Sign
     auto const pkSigner = ctx.tx.getSigningPubKey();
     if (!publicKeyType(makeSlice(pkSigner)))
     {
         JLOG(ctx.j.trace())
             << "checkSingleSign: signing public key type is unknown";
         return tefBAD_AUTH;  // FIXME: should be better error!
     }
-
-    // Look up the account.
     auto const idSigner = calcAccountID(PublicKey(makeSlice(pkSigner)));
-    auto const idAccount = ctx.tx.getAccountID(sfAccount);
     auto const sleAccount = ctx.view.read(keylet::account(idAccount));
     if (!sleAccount)
         return terNO_ACCOUNT;
 
+    return checkSingleSign(
+        idSigner, idAccount, sleAccount, ctx.view.rules(), ctx.j);
+}
+
+NotTEC
+Transactor::checkBatchSign(PreclaimContext const& ctx)
+{
+    NotTEC ret = tesSUCCESS;
+    STArray const& signers{ctx.tx.getFieldArray(sfBatchSigners)};
+    for (auto const& signer : signers)
+    {
+        auto const idAccount = signer.getAccountID(sfAccount);
+
+        Blob const& pkSigner = signer.getFieldVL(sfSigningPubKey);
+        if (pkSigner.empty())
+        {
+            STArray const& txSigners(signer.getFieldArray(sfSigners));
+            if (ret = checkMultiSign(ctx.view, idAccount, txSigners, ctx.j);
+                !isTesSuccess(ret))
+                return ret;
+        }
+        else
+        {
+            if (!publicKeyType(makeSlice(pkSigner)))
+                return tefBAD_AUTH;
+
+            auto const idSigner = calcAccountID(PublicKey(makeSlice(pkSigner)));
+            auto const sleAccount = ctx.view.read(keylet::account(idAccount));
+
+            // A batch can create an account ONLY when the account master key is
+            // the signer
+            if (!sleAccount)
+            {
+                if (idAccount != idSigner)
+                    return tefBAD_AUTH;
```

Comment:
Please add a test to cover this line

Reply:
Bump

Reply:
fixed: https://github.com/XRPLF/rippled/pull/5060/commits/94b403613bdf32afbc6bd7d1fe54ce1b7afc25ad

Reply:
reopening because codecov doesn't think it's covered